examples/platforms/efr32/radio.c

```diff
@@ -0,0 +1,839 @@
+                    error = kThreadError_NoBufs);
+
+    addToSrcMatchExtIndirect(entry, aExtAddress);
+
+exit:
+    CORE_EXIT_CRITICAL();
+    return error;
+}
+
+ThreadError otPlatRadioClearSrcMatchShortEntry(otInstance *aInstance, const uint16_t aShortAddress)
+{
+    ThreadError error = kThreadError_None;
+    int8_t entry = -1;
+    (void)aInstance;
+
+    CORE_DECLARE_IRQ_STATE;
+    CORE_ENTER_CRITICAL();
+
+    entry = findSrcMatchShortEntry(aShortAddress);
+    otLogDebgPlat(sInstance, "Clear ShortAddr entry: %d", entry);
+
+    otEXPECT_ACTION(entry >= 0 && entry < RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM,
+                    error = kThreadError_NoAddress);
+
+    removeFromSrcMatchShortIndirect(entry);
+
+exit:
+    CORE_EXIT_CRITICAL();
+    return error;
+}
+
+ThreadError otPlatRadioClearSrcMatchExtEntry(otInstance *aInstance, const uint8_t *aExtAddress)
+{
+    ThreadError error = kThreadError_None;
+    int8_t entry = -1;
+    (void)aInstance;
+
+    CORE_DECLARE_IRQ_STATE;
+    CORE_ENTER_CRITICAL();
+
+    entry = findSrcMatchExtEntry(aExtAddress);
+    otLogDebgPlat(sInstance, "Clear ExtAddr entry: %d", entry);
+
+    otEXPECT_ACTION(entry >= 0 && entry < RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM,
+                    error = kThreadError_NoAddress);
+
+    removeFromSrcMatchExtIndirect(entry);
+
+exit:
+    CORE_EXIT_CRITICAL();
+    return error;
+}
+
+void otPlatRadioClearSrcMatchShortEntries(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    otLogDebgPlat(sInstance, "Clear ShortAddr entries", NULL);
+
+    memset(srcMatchShortEntry, 0, sizeof(srcMatchShortEntry));
+}
+
+void otPlatRadioClearSrcMatchExtEntries(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    otLogDebgPlat(sInstance, "Clear ExtAddr entries", NULL);
+
+    memset(srcMatchExtEntry, 0, sizeof(srcMatchExtEntry));
+}
+
+void RAILCb_IEEE802154_DataRequestCommand(RAIL_IEEE802154_Address_t *aAddress)
+{
+    if (sIsSrcMatchEnabled)
+    {
+        if ((aAddress->length == RAIL_IEEE802154_LongAddress &&
+             findSrcMatchExtEntry(aAddress->longAddress) >= 0) ||
+            (aAddress->length == RAIL_IEEE802154_ShortAddress &&
+             findSrcMatchShortEntry(aAddress->shortAddress) >= 0))
+        {
+            RAIL_IEEE802154_SetFramePending();
+        }
+    }
+    else
+    {
+        (void)aAddress;
+        RAIL_IEEE802154_SetFramePending();
+    }
+}
+
+ThreadError otPlatRadioEnergyScan(otInstance *aInstance, uint8_t aScanChannel, uint16_t aScanDuration)
+{
+    (void)aInstance;
+    (void)aScanChannel;
+    (void)aScanDuration;
+    return kThreadError_NotImplemented;
+}
+
+void RAILCb_TxRadioStatus(uint8_t aStatus)
+{
+    switch (aStatus)
+    {
+    case RAIL_TX_CONFIG_CHANNEL_BUSY:
+        sTransmitError = kThreadError_ChannelAccessFailure;
+        sTransmitBusy = false;
+        break;
+
+    case RAIL_TX_CONFIG_TX_ABORTED:
+        sTransmitError = kThreadError_Abort;
+        sTransmitBusy = false;
+        break;
+    }
```

Comment:
According to RAIL documentation, there are some other return values here, but they are events which are not needed to notify the upper layer, for example, callback for CCA success, callback for when a CCA begins, callback for when CCA retries. When failure happens such as CCA failure (channel busy) or User abort Tx transmission, we should notify upper MAC layer to start a backoff method or retransmission. I can add a default status here for integrity.